src/main/java/io/citrine/jcc/search/core/result/BaseMultiSearchResult.java

```diff
@@ -55,17 +55,30 @@ public Long getTook() {
     }
 
     /**
+     * Set the list of successes.  This overwrites any successes that are already saved.
+     * @param successes List of successes.
+     * @return This object.
+     */
+    @JsonSetter("successes")
+    protected BaseMultiSearchResult<T> setSuccesses(final List<Boolean> successes) {
+        this.successes = successes;
+        return this;
+    }
+
+    /**
      * Add a single result that was returned.
      *
      * @param result Result to add to the results set.
      * @return This object.
      */
     @JsonIgnore
-    public BaseMultiSearchResult<T> addResult(final T result) {
+    public BaseMultiSearchResult<T> addResult(final T result, final Boolean success) {
```

Comment:
Java note: when you return `this`, you need to override the method in each subclass and have the subclass type returned. In this case, this applies to PifMultiSearchResult and DatasetMultiSearchResult - https://github.com/CitrineInformatics/java-citrination-client/blob/master/src/main/java/io/citrine/jcc/search/pif/result/PifMultiSearchResult.java. If you don't override and return the subclass, then when you chain methods together, you will get back the parent type even if operating on an instance of the subclass.